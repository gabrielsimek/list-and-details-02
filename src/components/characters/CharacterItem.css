li {
    list-style: none;
    /* background-color: lightslategray; */
    border: 1px solid black;
    padding: 15px;
    width: 200px;
    display: grid;
    grid-template-rows: 40px 160px 30px;
    grid-gap: 15px;
    justify-items: center;
}

li img {
    width: 100px;
    height: 100px;
    object-fit: contain;
}

li h1 {
    margin: 5px;
    font-size: 1.5rem;
}

