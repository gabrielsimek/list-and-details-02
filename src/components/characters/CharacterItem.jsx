import React from 'react';
import propTypes from 'prop-types';
import { Link } from 'react-router-dom';
import './CharacterItem.css';
const CharacterItem = ({ name, image, _id }) => {

  return (
    <Link to={`/character/${_id}`}> 
      <figure className="CharacterItem">
        <img 
          src={image}/>
        <figcaption>
          <h1>{name}</h1>
        </figcaption>
      </figure>
    </Link>
  );
 
};

CharacterItem.propTypes = {
  name: propTypes.string.isRequired,
  image: propTypes.string.isRequired,
  _id: propTypes.string.isRequired
};

export default CharacterItem;
