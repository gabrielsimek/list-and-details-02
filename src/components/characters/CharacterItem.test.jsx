import React from 'react';
import { render } from '@testing-library/react';

import CharacterItem from './CharacterItem';
import { MemoryRouter } from 'react-router-dom';


describe('App component', () => {
 
  it('renders App', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <CharacterItem 
          name="Mitzi"
          image="https://vignette.wikia.nocookie.net/heyarnold/images/d/d9/Mitzi.png/revision/latest/scale-to-width-down/310?cb=20171212070331"
          _id="5da237699734fdcb7bef8f54"
        />
      </MemoryRouter>);
    expect(asFragment()).toMatchSnapshot();
  });
});
