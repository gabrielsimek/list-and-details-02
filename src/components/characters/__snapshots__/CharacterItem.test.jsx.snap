// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders App 1`] = `
<DocumentFragment>
  <a
    href="/character/5da237699734fdcb7bef8f54"
  >
    <figure
      class="CharacterItem"
    >
      <img
        src="https://vignette.wikia.nocookie.net/heyarnold/images/d/d9/Mitzi.png/revision/latest/scale-to-width-down/310?cb=20171212070331"
      />
      <figcaption>
        <h1>
          Mitzi
        </h1>
      </figcaption>
    </figure>
  </a>
</DocumentFragment>
`;
